src/ActionInputValidator/Validations.ts

```diff
@@ -33,9 +33,9 @@ export function validateAppDetails() {
 
     if(!!actionParams.appName || (!!actionParams.slotName && actionParams.slotName !== 'production')) {
         let creds: ScmCredentials = PublishProfile.getPublishProfile(actionParams.publishProfileContent).creds;
-        let splitUsername: string[] = creds.username.substring(1).split("__");
-        let appNameMatch: boolean = !actionParams.appName || actionParams.appName === splitUsername[0];
-        let slotNameMatch: boolean = actionParams.slotName === 'production' || actionParams.slotName === splitUsername[1];
+        let splitUsername: string[] = creds.username.toUpperCase().substring(1).split("__");
+        let appNameMatch: boolean = !actionParams.appName || actionParams.appName.toUpperCase() === splitUsername[0];

```

Comment:
It is not really required to specify type for all cases. Typescript will auto detect the type in most cases.

Reply:
It does but this is best practice. typescript by definition allows you to define types for your variables. Just trying to use that capability explicitly